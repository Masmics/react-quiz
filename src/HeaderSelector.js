import React from 'react';
import PropTypes from 'prop-types';

function HeaderSelector({ headerText, headerChange }) {
  return (
    <input 
      name="headerText"
      type="text"
      value={headerText}
      onChange={headerChange}
    />
  );
}

HeaderSelector.propTypes = {
  headerText: PropTypes.string,
  headerChange: PropTypes.func
};

export default HeaderSelector;
