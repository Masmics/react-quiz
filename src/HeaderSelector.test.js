import React from 'react';
import { shallow } from 'enzyme';
import HeaderSelector from './HeaderSelector';

describe('header component', () => {
  it('allows input of header text', () => {
    const wrapper = shallow(<HeaderSelector />);
    expect(wrapper).toMatchSnapshot();
  });
});
