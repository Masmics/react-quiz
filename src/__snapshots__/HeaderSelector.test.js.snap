// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`header component allows input of header text 1`] = `
<input
  name="headerText"
  type="text"
/>
`;
